src/main/java/fr/insalyon/citi/golo/compiler/ParseTreeToGoloIrVisitor.java

```diff
@@ -35,6 +36,10 @@
 
   private GoloCompilationException.Builder exceptionBuilder;
 
+  private static String syntheticName(String base) {
+    return "__$$_" + base + "_" + System.currentTimeMillis();
```

Comment:
I'm not sure what exactly append here :wink:, but it seems to me possible to generate the same synthetic name both if we call this function two times in the same millisecond. Why not use an increment based on `Context` like with closure and decorators synthetic names?